pkg/controller/operators/operatorconditiongenerator_controller.go

```diff
@@ -0,0 +1,203 @@
+	}
+
+	csv := csvList.Items[0]
+
+	labelKey, err := operator.ComponentLabelKey()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// get all Service Accounts
+	serviceAccountList := corev1.ServiceAccountList{}
+	err = r.Client.List(context.TODO(), &serviceAccountList, client.InNamespace(csv.GetNamespace()))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	serviceAccounts := []string{}
+	for _, serviceAccount := range filterServiceAccounts(serviceAccountList.Items, hasOwnerFunc(ownerutil.NonBlockingOwner(&csv))) {
+		serviceAccounts = append(serviceAccounts, serviceAccount.GetName())
+	}
+	if len(serviceAccounts) == 0 {
+		serviceAccounts = []string{"default"}
+	}
+
+	// get all Service Accounts
+	deploymentList := appsv1.DeploymentList{}
+	err = r.Client.List(context.TODO(), &deploymentList, client.InNamespace(csv.GetNamespace()))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	deployments := []string{}
+	for _, deployment := range filterDeployments(deploymentList.Items, hasOwnerFunc(ownerutil.NonBlockingOwner(&csv))) {
+		deployments = append(deployments, deployment.GetName())
+	}
+
+	operatorCondition := operatorsv1.OperatorCondition{
+		ObjectMeta: metav1.ObjectMeta{
+			// For now, only generate an OperatorCondition with the same name as the csv.
+			Name:            csv.GetName(),
+			Namespace:       csv.Namespace,
+			Labels:          map[string]string{labelKey: ""},
+			OwnerReferences: []metav1.OwnerReference{ownerutil.NonBlockingOwner(&csv)},
+		},
+		Spec: operatorsv1.OperatorConditionSpec{
+			ServiceAccounts: serviceAccounts,
+			Deployments:     deployments,
+		},
+	}
+
+	err = r.Client.Create(context.TODO(), &operatorCondition)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			return ctrl.Result{Requeue: true}, err
+		}
+		existingOperatorCondition := &operatorsv1.OperatorCondition{}
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: operatorCondition.GetName(), Namespace: operatorCondition.GetNamespace()}, existingOperatorCondition)
+		if err != nil {
+			return ctrl.Result{Requeue: true}, err
+		}
+		existingOperatorCondition.OwnerReferences = operatorCondition.OwnerReferences
+		existingOperatorCondition.Spec = operatorCondition.Spec
+		err = r.Client.Update(context.TODO(), existingOperatorCondition)
+		if err != nil {
+			return ctrl.Result{Requeue: true}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func filterDeployments(deployments []appsv1.Deployment, f func(object metav1.Object) bool) []appsv1.Deployment {
+	// // filter based on OwnerReferences
+	filteredDeployments := []appsv1.Deployment{}
+	for _, deployment := range deployments {
+		if f(&deployment) {
+			filteredDeployments = append(filteredDeployments, deployment)
+		}
+	}
+	return filteredDeployments
+}
+
+func hasOwnerFunc(ownerReference metav1.OwnerReference) func(object metav1.Object) bool {
+	return func(object metav1.Object) bool {
+		return ownerutil.HasOwnerReference(object, ownerReference)
```

Comment:
there should be an `IsOwnedBy` already there

Reply:
I have removed the `owneruti.lHasOwnerReference` code in the latest version of the PR.